src/ipv4.c

```diff
@@ -515,12 +515,38 @@ int ipv4_protect_tunnel_route(struct tunnel *tunnel)
 	return ret;
 }
 
+static void add_text_route(struct tunnel *tunnel, const char *dest,
+                           const char *mask, const char *gw)
+{
+	size_t l0, l1;
+	const char fmt[] = ",%s/%s/%s";
+	const char trigger[] = "openfortivpn";
+	char **target = &tunnel->config->pppd_ipparam;
+	char *ptr;
+
+	if (*target == NULL || strncmp(*target, trigger, strlen(trigger)))
+		return;
+	if (!dest || !mask || !gw)
+		return;
+	log_info("Registering route %s/%s via %s\n", dest, mask, gw);
+	l0 = strlen(*target);
+	l1 = strlen(fmt) + strlen(dest) + strlen(mask) + strlen(gw) + 1;
+	if ((ptr = realloc(*target, l0 + l1))) {
+		*target = ptr;
+		snprintf(*target + l0, l1, fmt, dest, mask, gw);
+	} else {
+		int eno = errno;
```

Comment:
While log_error may theoretically change the value of `errno`, for mere consistency with the rest of the code I suggest getting rid of `eno` and using `errno` directly.

Reply:
Actually _log_error_() cannot change the value of `errno` before it is used as the `strerror(errno)` argument is evaluated before the function is called.